lib/sort_by_length.rb

```diff
@@ -1,7 +1,35 @@
 # A method which will return an array of the words in the string
 #  sorted by the length of the word.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n^2) insertion sort
+# Space complexity: 0(n) created new array&hash that have the linear relationship with the length of the input
 def sort_by_length(my_sentence)
```

Comment:
Nice insertion sort, a bit of extra work put in here, but it does work.  You don't really need a hash I think.